packages/vite/src/node/optimizer/scan.ts

```diff
@@ -385,7 +385,10 @@ function esbuildScanPlugin(
       build.onLoad(
         { filter: htmlTypesRE, namespace: 'html' },
         async ({ path }) => {
-          let raw = await fsp.readFile(path, 'utf-8')
+          let raw = isVirtualModule(path)
+            ? await container.load(path)
```

Comment:
The optimizer support plugin `load` is complex, I think using `external` for the virtual module is a simple idea.